core/services/ping/pingmanager.py

```diff
@@ -1,38 +1,68 @@
+import asyncio
 import logging
-from typing import Dict, List
+from typing import Any, Dict, List
 
 from serial import Serial
 
 from ping1d_driver import Ping1DDriver
 from ping360_driver import Ping360Driver
 from pingdriver import PingDriver
-from pingutils import PingDeviceDescriptor, PingType
+from pingutils import PingDeviceDescriptor, PingType, udp_port_is_in_use
 
 
 class PingManager:
     def __init__(self) -> None:
         self.drivers: Dict[PingDeviceDescriptor, PingDriver] = {}
-        self.ping1d_current_port: int = 9090
-        self.ping360_current_port: int = 9092
+        self.ping1d_base_port: int = 9090
+        self.ping360_base_port: int = 9092
 
     def stop_driver_at_port(self, port: Serial) -> None:
         """Stops the driver instance running for port "port" """
-        ping_at_port = list(filter(lambda ping: ping.port == port, self.drivers.keys()))
+        ping_at_port = [ping for ping in self.drivers if ping.port == port]
         if ping_at_port:
             self.drivers[ping_at_port[0]].stop()
+            del self.drivers[ping_at_port[0]]
 
-    def launch_driver_instance(self, ping: PingDeviceDescriptor) -> None:
+    async def register_ethernet_ping360(self, ping: PingDeviceDescriptor) -> None:
+        if ping not in self.drivers:
+            self.drivers[ping] = Ping360Driver(ping, None)
+
+    async def find_next_port(self, base_port: int, direction: int) -> int:
+        """
+        Find the next unused udp port, starts on base_port and increments/decrements
+        accordingly to direction
+        """
+        port = base_port
+        while udp_port_is_in_use(port):
```

Comment:
We should limit the range that the port can be.
At least between 9090 and 9099, otherwise we could start having numbers totally outside the range if 3rd party programs are using it, without us having control to find the sensors with ping-viewer.
We should also start thinking about how ping-viewer is going to discover this devices, how that works currently ?

Reply:
I don't see the point in limiting it. Discovery needs to be improved in PingViewer.
As is it will work fine for one Ping1D and one Ping360.
We could either use the Ping360 discovery protocol, or ask Beacon to broadcast the devices using mDNS

Reply:
Not limiting the range will result in a silent loop block where there is no condition to exit besides the OS saying that a port is available, if no port is available, this loop will never exit. But if you don't have strong feelings that such thing will never happen I'm ok with it.